src/tiled/terrainbrush.cpp

```diff
@@ -371,6 +372,26 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         int x = p.x(), y = p.y();
         int i = y*layerWidth + x;
 
+        // to support isometric staggered, make edges into variables
+        QPoint upPoint(p.x(),p.y()-1);
+        QPoint bottomPoint(p.x(),p.y()+1);
+        QPoint leftPoint(p.x()-1,p.y());
+        QPoint rightPoint(p.x()+1,p.y());
```

Comment:
I just realized there are already local variables `x` and `y` for the current point, so please replace all `p.x()` and `p.y()` with that for brevity.